### PR TITLE
Support OCaml 4.04

### PR DESCRIPTION
**Pull request by anmonteiro**
If H2 is ever going to use this implementation, it needs to support OCaml 4.04.

### CODE REVIEW

**Comment by 314eter** [comment]
I didn't think about that yet. Maybe it's easier to replace
```ocaml
begin match ValueMap.find_opt value map with
| Some seq -> ...
| None -> ...
end
```
by
```ocaml
begin match ValueMap.find value map with
| Some seq -> ...
| exception Not_found -> ...
end
```
instead of extending the module (and the same for `Hashtbl`).

File: util/gen_static.ml [hunk 1]
```diff
@@ -1,34 +1,36 @@
...
-      |> CharSet.of_seq
+      StringSet.elements names
+      |> List.map (fun name -> name.[pos])
+      |> CharSet.of_list
       |> CharSet.cardinal
       |> ( = ) n
     then pos else loop (pos + 1) in
   loop 0
 
+let make_token_map static_table =
```
Comment (314eter) [inline util/gen_static.ml]: Maybe `token_table` is a better name now that it's not a map anymore.

File: src/encoder.ml [hunk 2]
```diff
@@ -98,33 +99,40 @@ let encode ({lookup_table; next_seq; _} as encoder) {name; value; never_index} =
...
   else
-    match token, LookupTable.find_opt lookup_table name with
-    | Some token, Some map ->
-      begin match ValueMap.find_opt value map with
-      | Some seq -> Index (seq_to_index next_seq seq)
-      | None -> find_token encoder no_index token name value
+    begin match token with
+    | Some token ->
+      begin match LookupTable.find lookup_table name with
+      |  map ->
```
Comment (314eter) [inline src/encoder.ml]: There is an extra space before `map`.

**Comment by 314eter** [comment]
Very nice, you can merge this as soon as the extra space is removed and the tests are green.

File: src/encoder.ml (hunk 2 above)
Comment (anmonteiro) [reply src/encoder.ml]: ah indeed. removed `Some` and kept the space.

**Pull request merged**